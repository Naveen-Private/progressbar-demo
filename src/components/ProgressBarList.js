import React from 'react'
import ProgressBar from './ProgressBar'

const ProgressBarList = (props) => {
    return(
        props.progressBars.map((progressBar, index) =>
            <ProgressBar key={index} value={progressBar} />
        )
    )
}

export default ProgressBarList
