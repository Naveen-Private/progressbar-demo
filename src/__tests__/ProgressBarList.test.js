import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ProgressBarList from '../components/ProgressBarList'

describe('ProgressBarList', () => {
    it('should be defined', () => {
      expect(ProgressBarList).toBeDefined();
    });
    it('should render correctly', () => {
      const tree = shallow(
        <ProgressBarList progressBars={[10, 20, 30]} />
      );
      expect(tree).toMatchSnapshot();
    });
});