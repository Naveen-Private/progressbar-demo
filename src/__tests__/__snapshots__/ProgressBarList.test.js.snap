// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBarList should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBarList
    progressBars={
        Array [
            10,
            20,
            30,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <ProgressBar
      value={10}
/>,
    <ProgressBar
      value={20}
/>,
    <ProgressBar
      value={30}
/>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <ProgressBar
        value={10}
/>,
      <ProgressBar
        value={20}
/>,
      <ProgressBar
        value={30}
/>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
